### ISSUE TITLE
Changing project from sidebar overwrites it's path in projects.cson

### ISSUE THREAD

**Issue by redfellow** [bripkens/project-sidebar#8]
```
projectname:
  title: "title here"
  paths: [
    "/path/here"
  ]
```

The /path/here inside paths changes when I click a project in the sidebar, which "corrupts" the file as if you close atom, work on another project and then try to open the first project again, it's path is whatever the last project you worked on were.

Chaging project via the regular project manager does not do this.

edit: Ubuntu 14.04, project-manager 2.6.5 and project-sidebar 1.3.0 & atom 1.0.19

**Comment by bripkens** [comment]
Do you mean that the `projects.cson` file gets corrupted?

**Comment by redfellow** [comment]
Yeah, if I keep the file open while clicking a project, I can see the "path/here" part change every time I click a project via the sidebar.

**Comment by bripkens** [comment]
That is unusual. Unfortunately for you, `project-sidebar` does not change the `projects.cson` file. You can verify this by inspecting the [Sidebar.es6](https://github.com/bripkens/project-sidebar/blob/b65761c36941c2e24a501bbfc259cca5df70171b/src/Sidebar.es6#L41-L44). The logic is quite simple, so I am afraid I don't know how to help you :-(.

**Comment by redfellow** [comment]
Weird, the only project-plugins I have installed are project-manager and project-sidebar, and the only place this weird thing happens is if I change project via the sidebar, as project-managers own toggle loads a new window.

**Comment by bripkens** [comment]
Does project-manager nowadays synchronize the open projects with the `projects.cson` file? That might explain this behavior.

**Comment by redfellow** [comment]
```
// remove all current directories that do not belong to the
// selected project.
```

Was looking bit below from the file you linked, and was wondering if this could be the culprit. Modifying open directories on the fly might explain why the cson gets updated. project-manager seems to instead load selected project in a new window without touching the tree.

**Comment by bripkens** [comment]
Yes, `project-sidebar` modifies the current window's open project. Though this is not synchronized with `project.cson` by `project-sidebar`. This is why I was suspecting a synchronization feature in `project-manager`.

**Comment by eliot-akira** [comment]
Had the same issue: switching projects overwrites the current project's folders in _projects.cson_. Didn't notice it for a while, until I realized many projects' folders were mixed up due to this.

Tracked it down to `project-manager`, indeed [it syncs _projects.cson_ when the current project's paths change](https://github.com/danielbrodin/atom-project-manager/blob/master/lib/project-manager.js#L59). For now, I've commented out that line to prevent this issue. I can always save projects manually if I add/remove folders.

**Comment by jonchenn** [comment]
Same here.  Thanks for ellot-akira@'s quick fix solution. It works fine now. Perhaps it's an edge case for project-manager package?

**Comment by mehcode** [comment]
This is because `atom.project.setPaths` is not the correct method to switch to a different logical project / path set. It should only be used to adjust/edit the paths for the current project (which explains `project-manager`'s behavior). There is no exposed API to do the latter.

---

Now the latter can be done but it can get nasty. It's gotten a _lot_ better as of 1.7.0-beta (~10 lines vs. ~100). This is implemented in https://github.com/mehcode/atom-project-plus for those that are interested.
